Add agent framework/throttling/hidden model/OS assistant and update conversational search documentation (#6354)
opensearch-project/documentation-website: _ml-commons-plugin/agents-tools/tools/search-anomaly-results.md

```json
{
  "inference_results": [
    {
      "output": [
        {
          "name": "response",
          "result": "AnomalyResults=[{detectorId=ef9lYo0Bk4MTqircmjnm,grade=1.0,confidence=0.9403051246569198}{detectorId=E-JlYo0B9RaBCvhunFtw,grade=1.0,confidence=0.9163498216870274}]TotalAnomalyResults=2"
        }
      ]
    }
  ]
}
```

If no anomalies are found, OpenSearch responds with an empty array in the results:

```json
{
  "inference_results": [
    {
      "output": [
        {
          "name": "response",
          "result": "AnomalyResults=[]TotalAnomalyResults=0"
        }
      ]
    }
  ]
}
```

## Parameters

The following table lists all available parameters. All parameters are optional.

Parameter	| Type | Description	
:--- | :--- | :---
`detectorId`	| String	| The ID of the detector whose results to return.
`realTime`	| Boolean | Whether to return real-time anomaly detector results. Set this parameter to `false` to return only historical analysis results.

> `anomalyGradeThreshold` | Float	| The minimum anomaly grade for anomaly detector results to return. Anomaly grade is a number between 0 and 1 that indicates how anomalous a data point is.

Author: ```suggestion
`anomalyGradeThreshold` | Float	| The minimum anomaly grade for the returned anomaly detector results. Anomaly grade is a number between 0 and 1 that indicates how anomalous a data point is.
```

`dataStartTime` | Long	| The earliest time for which to return anomaly detector results, in epoch milliseconds.
`dataEndTime` | Long |	The latest time for which to return anomaly detector results, in epoch milliseconds.

> `sortOrder`	|String | The sort order for the results. Valid values are `asc` (ascending) and `desc` (descending). Default is `desc`. 

Author: ```suggestion
`sortOrder`	|String | The sort order of the results. Valid values are `asc` (ascending) and `desc` (descending). Default is `desc`. 
```

`sortString`| String |	Specifies the detector field by which to sort the results. Default is `data_start_time`.

> `size`	| Integer |	The number of results to return. Default is 20.

Author: ```suggestion
`size`	| Integer |	The number of results to return. Default is `20`.
```

> `startIndex`| Integer |	The paginated index of the result to start from. Default is 0.

Author: ```suggestion
`startIndex`| Integer |	The paginated index of the result to start from. Default is `0`.
```
